pkg/flow/internal/controller/loader.go

```diff
@@ -392,6 +402,10 @@ func (l *Loader) EvaluateDependencies(parentScope *vm.Scope, c *ComponentNode) {
 		switch n := n.(type) {
 		case BlockNode:
 			err = l.evaluate(logger, parentScope, n)
+			if exp, ok := n.(*ExportConfigNode); ok {
+				name, val := exp.NameAndValue()
+				exports[name] = val
```

Comment:
I thought for a moment that we should validate if the key is already here but we have an upstream check on duplicate blocks that will catch this. Mentioning in case anyone thinks similar.

Example error:
<img width="763" alt="image" src="https://user-images.githubusercontent.com/39704712/229147536-93a161fa-0a0b-44a3-8bc1-169514f39ab3.png">